protocol-units/bridge/contracts/src/AtomicBridgeInitator.sol

```diff
@@ -0,0 +1,96 @@
+            revert ZeroAddress();
+        }
+        weth = IWETH9(_weth);
+    }
+
+    function initiateBridgeTransfer(uint256 wethAmount, bytes32 recipient, bytes32 hashLock, uint256 timeLock)
+        external
+        payable
+        returns (bytes32 bridgeTransferId)
+    {
+        address originator = msg.sender;
+        uint256 ethAmount = msg.value;
+        uint256 totalAmount = wethAmount + ethAmount;
+        // Ensure there is a valid total amount
+        if (totalAmount == 0) {
+            revert ZeroAmount();
+        }
+        // If msg.value is greater than 0, convert ETH to WETH
+        if (ethAmount > 0) weth.deposit{value: ethAmount}();
+        //Transfer WETH to this contract, revert if transfer fails
+        if (wethAmount > 0) {
+            if (!weth.transferFrom(originator, address(this), wethAmount)) revert WETHTransferFailed();
+        }
+
+        nonce++; //increment the nonce
+        bridgeTransferId =
+            keccak256(abi.encodePacked(originator, recipient, hashLock, timeLock, block.timestamp, nonce));
+
+        // Check if the bridge transfer already exists
+        if (bridgeTransfers[bridgeTransferId].amount != 0) revert BridgeTransferInvalid();
+
+        bridgeTransfers[bridgeTransferId] = BridgeTransfer({
+            amount: totalAmount,
+            originator: originator,
+            recipient: recipient,
+            hashLock: hashLock,
+            timeLock: block.timestamp + timeLock,
+            completed: false
+        });
+
+        bridgeTransferIds.push(bridgeTransferId);
+
+        emit BridgeTransferInitiated(bridgeTransferId, originator, recipient, totalAmount, hashLock, timeLock);
+        return bridgeTransferId;
+    }
+
+    function completeBridgeTransfer(bytes32 bridgeTransferId, bytes32 preImage) external {
+        // Retrieve the bridge transfer
+        BridgeTransfer storage bridgeTransfer = bridgeTransfers[bridgeTransferId];
+        if (bridgeTransfer.completed) revert BridgeTransferHasBeenCompleted();
+        if (keccak256(abi.encodePacked(preImage)) != bridgeTransfer.hashLock) revert InvalidSecret();
+        bridgeTransfer.completed = true;
+        emit BridgeTransferCompleted(bridgeTransferId, preImage);
+    }
+
+    function refundBridgeTransfer(bytes32 bridgeTransferId) external {
+        BridgeTransfer storage bridgeTransfer = bridgeTransfers[bridgeTransferId];
+        uint256 amount = bridgeTransfer.amount;
+        if (bridgeTransfer.completed) revert BridgeTransferHasBeenCompleted();
+        if (block.timestamp < bridgeTransfer.timeLock) revert TimeLockNotExpired();
+
+        bridgeTransfer.completed = true;
+
+        if (!weth.transfer(bridgeTransfer.originator, amount)) revert WETHTransferFailed();
+        emit BridgeTransferRefunded(bridgeTransferId);
+    }
+
+    // Get all bridge transfer ids, the solidity get method only returns one item by index
+    function getAllBridgeTransferIds() external view returns (bytes32[] memory) {
```

Comment:
Opting for this solution might initially seem adequate; however, it is important to reconsider our approach to working with the completed component, as maintaining it could lead to unsustainable scalability in the long term. Consequently, we must carefully evaluate the maximum return limits to prevent potential errors.

Reply:
I'm not sure there is a max return limit? Or if there is, we would never hit that limit, not infinite but a very large number

Reply:
As discussed, remove as indexing will be done via blockheigh, replaying history and event listening. No need for this funciton anymore